### ISSUE TITLE
Document RDS rollback procedure

### ISSUE THREAD

**Issue by atolopko-czi** [chanzuckerberg/single-cell-data-portal#4499]
## Motivation

Engineers do not have guidelines on how to rollback a prod deployment. If a prod deployment is broken and had RDS database migrations applied, rolling back requires rolling back the migrations as well. But it is not immediately obvious how to do this.

## Definition of Done

* All engineers know how to rollback a RDS migration after a failed deployment, using a  documented playbook.
* All engineers can quickly determine _whether_ an RDS rollback is necessary for a given deployment.
* The RDS migration rollback procedure is tested in a non-prod env.

## Tasks

* Document here](https://docs.google.com/document/d/1n5cngEIz-Lqk9737zz3makXGTMrEKT5kN4lsofXPRso/edit#heading=h.dujy4xt9xsik) how to perform deployment rollbacks:
    * How to deploy the previous Docker image build.
    * How to run a reversible alembic "down" migrations (only needed for prod, staging can be re-mirrored).
    * How to restore database from a snapshot using AWS console and/or AWS CLI commands  (only needed for prod and only if alembic "down" migration is not reversible.)
* Update [Single Cell On-call Log/Checklist](https://docs.google.com/document/d/1G2NTjXTJJeHyhqvnyzYmcO0Um24Ph0dCLUyMIWZvLfg/edit) checklists:
    * Staging and Prod checklists should include a pre-deployment step to determine whether an RDS migration will occur, asking the engineer to note the status in the deployment log: "no migration", "reversible migration", "non-reversible-migration".
    * Prod checklist should also include a step (w/instructions) to take a snapshot of the database _if_ the database migration is not reversible, noting the snapshot ARN in the deployment log.

**Comment by ebezzi** [comment]
Personally I don't think this is very useful. One of the goal of portal redesign was to minimize the necessity of database migrations by moving the fields that require evolution to JSON fields. I'm expecting most (if not all) of the schema migrations to be field additions; these generally don't need to be rolled back as older versions of the code will just ignore the new field.

I also would like to stress that database migrations that aren't backwards compatible (i.e. the ones that need to be rolled back) cause downtime and therefore should be used very sparingly. I don't see how a "generic" oncall person (i.e. someone who has no context of the migration process) should need to perform a complicated process like a RDS rollback.

**Comment by atolopko-czi** [comment]
> I also would like to stress that database migrations that aren't backwards compatible (i.e. the ones that need to be rolled back) cause downtime and therefore should be used very sparingly.

I agree and I'll add that a rollback should only be performed if we're already experiencing downtime (full or partial, say, at a feature level).
